dashboard/app/controllers/api/v1/schools_controller.rb

```diff
@@ -10,11 +13,9 @@ def index
 
   # GET /dashboardapi/v1/schoolsearch/:q/:limit
   def search
+    query = params.require(:q)
     limit = [40, Integer(params[:limit])].min
-    schools = School.where("name LIKE ?", "%#{params[:q]}%").order(:name).limit(limit)
-    serialized_schools = schools.map do |school|
-      Api::V1::SchoolAddressSerializer.new(school).attributes
-    end
-    render json: serialized_schools
+    render json: SchoolAutocomplete.instance.get_matches(query, limit)
```

Comment:
This will, per your timing, have a delay of about 15 seconds the first time it is hit on each front end. This is OK to ship for now while it is not getting used. Before we use this in production we should warm this up at app initialization like we do with levels cache.